hatasmota/camera.py

```diff
@@ -0,0 +1,131 @@
+"""Tasmota camera."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from aiohttp import ClientResponse, ClientSession
+
+from .const import CONF_DEEP_SLEEP, CONF_IP, CONF_MAC
+from .entity import (
+    TasmotaAvailability,
+    TasmotaAvailabilityConfig,
+    TasmotaEntity,
+    TasmotaEntityConfig,
+)
+from .mqtt import ReceiveMessage
+from .utils import (
+    config_get_state_offline,
+    config_get_state_online,
+    get_topic_command,
+    get_topic_command_state,
+    get_topic_stat_result,
+    get_topic_tele_sensor,
+    get_topic_tele_state,
+    get_topic_tele_will,
+    get_value_by_path,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class TasmotaCameraConfig(TasmotaAvailabilityConfig, TasmotaEntityConfig):
+    """Tasmota camera configuation."""
+
+    idx: int
+
+    command_topic: str
```

Comment:
This doesn't seem to be used

Reply:
It was added earlier to support the configuring the camera parameters. But since it is not used, I have removed it.
The configuration is done easily via the blueprint mentioned earlier.

Reply:
Yes. A number entity for resolution, brightness, and contrast.
We also have switch entities for flip horizontal/vertical.

Overall it looks like this with my [blueprint](https://github.com/tasmota/blueprints/pull/4):
![image](https://github.com/user-attachments/assets/ea8d4e8b-0a3e-4547-8433-ec68644a12ee)

Reply:
OK. You could add this support to hatasmota and the tasmota integration if you want, as number entities and switch entities which are discovered if there's a camera.